mycroft/skills/core.py

```diff
@@ -459,6 +459,10 @@ def register_intent(self, intent_parser, handler, need_self=False):
         elif type(intent_parser) != Intent:
             raise ValueError('intent_parser is not an Intent')
 
+        if not intent_parser.name:
+            # Default to the handler's function name if None or ""
+            intent_parser.name = handler.__name__
+
         name = intent_parser.name
         intent_parser.name = str(self.skill_id) + ':' + intent_parser.name
```

Comment:
This could be a bit more compact and just as easy to understand:

```python
        # Default to the handler's function name if None or ""
        name = intent_parser.name or handler.__name__
        intent_parser.name = str(self.skill_id) + ':' + name
```

Reply:
Done.  You are so Pythonic.  ;-)